Makefile

```diff
@@ -55,11 +52,21 @@ help: ## Display this help.
 	@printf "$$(git rev-parse HEAD 2>/dev/null)" >$@
 	@test -n "$$(git status --porcelain --untracked-files=no)" && echo -dirty >>$@ || true
 
-agent-protocol-forwarder: .git-commit $(SOURCES)
-	$(GOOPTIONS) CGO_ENABLED=0 go build $(GOFLAGS) -ldflags="-X 'github.com/confidential-containers/cloud-api-adaptor/cmd.VERSION=$(shell echo "unknown")' -X 'github.com/confidential-containers/cloud-api-adaptor/cmd.COMMIT=$(shell cat .git-commit)'" -o "$@" "cmd/$@/main.go"
+GOFLAGS += -ldflags="-X 'github.com/confidential-containers/cloud-api-adaptor/cmd.VERSION=$(shell echo "unknown")' \
+                     -X 'github.com/confidential-containers/cloud-api-adaptor/cmd.COMMIT=$(shell cat .git-commit)'"
 
-cloud-api-adaptor: .git-commit $(SOURCES)
-	$(GOOPTIONS) go build $(GOFLAGS) -ldflags="-X 'github.com/confidential-containers/cloud-api-adaptor/cmd.VERSION=$(shell echo "unknown")' -X 'github.com/confidential-containers/cloud-api-adaptor/cmd.COMMIT=$(shell cat .git-commit)'" -o "$@" "cmd/$@/main.go"
+# Build tags required to build cloud-api-adaptor are deribed from BUILTIN_CLOUD_PROVIDERS.
+# When libvirt is specified, CGO_ENABLED is set to 1.
+space := $() $()
```

Comment:
Can you please explain this ? Is it any number of spaces between the provider strings ?

Reply:
This is the technique described in the GNU Make manual. 
https://www.gnu.org/software/make/manual/html_node/Syntax-of-Functions.html

The number of spaces between provider strings should be one. I don't have an idea to handle consecutive spaces in Makefile. 

If BUILTIN_CLOUD_PROVIDERS contains two consecutive spaces,
```
BUILTIN_CLOUD_PROVIDERS ?= aws azure ibmcloud  vsphere
```

the tag option looks like this.
```
-tags aws,azure,ibmcloud,,vsphere
```

I confirm that `go build` does not complain about this empty build tag, so there is no real harm here.

Reply:
Would it be better to have the BUILTIN_CLOUD_PROVIDERS as a comma separated list ?
I have no preferences. Trying to figure out least error prone approach when users want to set custom `BUILTIN_CLOUD_PROVIDERS` before build.

Reply:
I noticed that the `scrip` function replaces each consecutive spaces with a single space. I'll use this function to fix the problem.
https://www.gnu.org/software/make/manual/html_node/Text-Functions.html#index-stripping-whitespace

Reply:
@bpradipt updated